### ISSUE TITLE
https redirect with custom domain on github pages and cloudflare

### ISSUE THREAD

**Issue by wAuner** [mmistakes/minimal-mistakes#1614]
## Description

* hosting: github pages, private repo
* custom domain
* domain is routed through cloudflare, to enable https for github+custom domain roughly based on [this post](https://blog.cloudflare.com/secure-and-fast-github-pages-with-cloudflare/)
* domain in _config.yml: `url                      : https://winfriedauner.de`

It almost works as expected. However, when I type `winfriedauner.de` I get the website without https. Once I click on any section in the navigation, https works. When I try to activate a redirection in cloudflare (as suggested by cloudflare and other tutorials) of http://winfriedauner.de to the https version, I get an error of too many redirections and the site doesn't load. 

Is there something else I need to adapt?
Thanks!

## Environment

- Minimal Mistakes version: updated today, so 4.11.1 I believe
- Ruby gem or remote theme version: remote theme
- Jekyll version: github
- Git repository URL: private repo
- Operating system: Linux Mint
- GitHub Pages hosted (if yes provide URL to site):

---

**Comment by justinrummel** [comment]
I don't think Cloudflare is setup correctly.  I can still access http://winfriedauner.de/about/ without getting a 301 redirect to the HTTPS version. 

``` bash
jrummel@res-mpiff ~> curl -sI "http://winfriedauner.de/cv/" | grep "HTTP/\|Location"
HTTP/1.1 200 OK
```

I don't see your MM repo on your GitHub account, so can't review your ```_config.yml``` either.  I self host, but run my site through cloudflare and everything is forced as HTTPS.

**Comment by mmistakes** [comment]
@wAuner I would suggest you open up your question to the larger Jekyll community to see if others have had success integrating GitHub Pages with Cloudflare.

This has little to do with Jekyll... and even less with the theme. It's purely a hosting thing. You could try asking on [Jekyll Talk](https://talk.jekyllrb.com/), but likely would have greater success with [Stackoverflow](https://stackoverflow.com/search?q=cloudflare+github).

**Comment by wAuner** [comment]
@mmistakes the problem is that no one else seem to have this problem. Everyone else is just able to set up redirects and it works. Others don't seem to have this problem with Jekyll. However, my site doesn't render correctly or isn't available. That's why I thought I might be due to the theme I use.

**Comment by tony-ho** [comment]
@wAuner Were you trying to redirect to HTTPS using Cloudflare Page Rules? I suspect you may have had conflicting rules, which caused the redirect loop.

The easiest way to force HTTPS site-wide is turn on **Always Use HTTPS** in the Crypto section of the Cloudflare Dashboard:
https://blog.cloudflare.com/how-to-make-your-site-https-only/

This is how my (previously Minimal Mistakes) GitHub Pages with Cloudflare site is set up.

**Comment by wAuner** [comment]
@tony-ho hey tony, thanks. I was certain I tested that already, but not probably not without deactivating all the rules. It seems to work now. I've deactivated all redirections in cloudflare and just activated "always use https". Thank you! 
I also deactivated the "remote theme" option again. That caused issues when I was testing offline and doesn't seem to make any difference when I host it on github.
Thanks @everyone for your help!
